doc/provisioning.md

```diff
@@ -1,19 +1,28 @@
-What We Run, and Why
-====================
-
-Expressed as compactly as possible, the Provisioner is responsible for making
-sure that non-Dead machine entities in state have agents running on live
-instances; and for making sure that Dead machines, and stray instances, are
-removed and cleaned up.
-
-However, the choice of exactly what we deploy involves some subtleties. At the
-Provisioner level, it's simple: the base and the constraints we pass to the
-Environ.StartInstance come from the machine entity. But how did they get there?
-
-Bases
------
-
-Individual charms are released for different possible target base; juju
+# Provisioning
+
+Provisioning in Juju is performed by two separate workers:
+- The [compute provisioner](https://github.com/juju/juju/blob/main/internal/worker/computeprovisioner/doc.go),
+which is responsible for provisioning machine instances on the underlying
+provider.
+- The [container provisioner](https://github.com/juju/juju/blob/main/internal/worker/containerprovisioner/doc.go),
+which is responsible for creating containers managed by the container
+broker on the underlying machine.
+
+Juju splits the resource creation in two stages. When a new machine
+is being created (either by the user explictly or because a new unit
+is deployed) the machine is first stored in the database, and then
+one of the provisioners (compute or container) workers will pick it
+up and start the provisioning process. Both workers will gather all the
+necessary information and then create a [provisioner_task](https://github.com/juju/juju/blob/main/internal/provisionertask/provisioner_task.go)
+worker which contains all the bussiness logic for creating and updating
+instances, as well as keeping track of all the instances.
+
+The following sections will describe in more detail the different
+pieces that participate in the provisioning process.
+
+## Bases
+
+Individual charms are released for different possible target bases; Juju
 should guarantee that charms for base X are only ever run on base X.
 Every service, unit, and machine has a base that's set at creation time and
```

Comment:
Its unclear what service means here. Is it application? I know you didn't write this but it might be good to either define it or change it.

Reply:
I replaced service by application, which makes sense in today's Juju anyways.